src/react/fileUrlToUrl.ts

```diff
@@ -89,7 +93,9 @@ export const getTranslatedPathPattern = ({
   // if nextPathPart does not match any child name and a dynamic child is found,
   // we will consider that nextPathPart is a value given to the dynamic child
 
-  const matchingChild = routeBranch.children?.find((child) => child.name === nextFilePathPart)
+  const matchingChild = routeBranch.children?.find(
+    (child) => child.name === nextFilePathPart || child?.paths?.[locale] === nextFilePathPart,
```

Comment:
Since `nextFilePathPart` is a **file** path part, it is not translated: no need to check a match with a translation, only a match with the branch name is needed.

Reply:
If we translate the paths in `fileUrlToFileUrlObject` then we need to correctly find the matching child by locale otherwise the translation won't work you can try removing that line yourself and seeing how the rest breaks as you can't match translated paths with default ones 🙃